### PR TITLE
Amend ADR004 to allow extracting forms during the auth phase

### PR DESCRIPTION
**Pull request by balloob**
We've seen more and more integrations struggle with ADR004. The problem they struggle with is not that the data fetching is based on scraping, but that some parts of the authentication flow requires extraction of the form fields and their values. Fetching data should happen via an API. We will be more flexible, and hinder less integrations, if we allow integrations to extract forms during the authentication flow. 

That's why I want to propose that we amend ADR004 to allow web scraping of forms during the authentication phase. For robustness, integrations should fetch all fields of the form and not individually extract values of fields in the form. These are less likely to change.

Please limit discussion in this PR to this change. Discussing the ADR004 or other amendments is out of scope for this PR. Start a discussion topic instead.

### CODE REVIEW

File: adr/0004-webscraping.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # 4. Webscraping
-
+gs
```
Comment (MartinHjelmare) [inline adr/0004-webscraping.md:2]:
```suggestion
```

Comment (frenck) [inline adr/0004-webscraping.md:2]: ?

Comment (balloob) [reply adr/0004-webscraping.md:2]: haha oops that's me thinking I'm in my terminal 😬

**Comment by frenck** [comment]
I think this is fine. Should we add an update date or changelog or something to these?

**Comment by balloob** [comment]
Added a changelog.

**Review by frenck — APPROVED** [review]

**Review by pvizeli — APPROVED** [review]

File: adr/0004-webscraping.md [hunk 2]
```diff
@@ -25,6 +29,10 @@ Webscraping comes with the following downsides:
 - It will still be possible to have custom integrations provide information via webscraping
 - Generic integration to parse HTML are excluded from this decision
 
+### Exception
```
Comment (ludeeus) [inline adr/0004-webscraping.md:32]: We already have an exception "Generic integration to parse HTML are excluded from this decision"
Should not the new "Exception" section should include both?

**Review by ludeeus — APPROVED** [review]

**Review by bramkragten — APPROVED** [review]

**Review by bdraco — APPROVED** [review]

**Review by bdraco — APPROVED** [review]

**Review by raman325 — APPROVED** [review]

**Review by rohankapoorcom — APPROVED** [review]

**Review by KimGuang44 — APPROVED** [review]
感恩您，煩請下次更改時附註一封中文信件說明修改了什麼，謝謝

**Review by alandtse — APPROVED** [review]

**Comment by amelchio** [comment]
I am fine with the change. However, an ADR should not just document _what_ has been decided but also _why_.

In #270 webscraping of local devices was rejected. Top arguments included webscraping being a horror that must unconditionally be avoided and the observation that a local website is still a webstite. This document should make it clear to future contributors (and maybe me) why those concerns do not apply to login pages.

The reasoning in the description does not seem to explain it, it sounds more like a contradiction to me:

> The problem is not that the API is based on scraping, but that some parts of the login form require extraction of all the form fields.

**Comment by balloob** [comment]
I've updated the initial comment.

**Pull request merged**